sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,191 @@
+
+kubevirt/kubevirt.
+
+# Design
+
+## API Addition
+Currently, the `Interface` struct has the `State` field (currently used for hot-unplug):
+```go
+type Interface struct {
+    ...
+    State InterfaceState `json:"state,omitempty"`
+    ...
+}
+```
+
+Type `InterfaceState` is an enum, currently having a single value:  
+```go
+type InterfaceState string
+
+const (
+	InterfaceStateAbsent InterfaceState = "absent"
+)
+```
+
+The following values will be added to specify the required link state:
+```go
+const (
+    InterfaceStateLinkUp   InterfaceState = "linkUp"
+    InterfaceStateLinkDown InterfaceState = "linkDown"
+)
+```
+
+An empty value will be considered as `linkUp`.
+
+The network validator will be adjusted to allow the two new values for:
+1. All interfaces connected to primary and secondary networks.
+2. All bindings - core and plugins.
+
+virt-launcher's `Converter` component will be adjusted to take the interface State field into account when creating a new domain XML.
+In case the field's value is `InterfaceStateLinkDown`, the interface will be created as follows:
+```xml
+<interface>
+  ...
+  <link state='down'/>
+  ...
+</interface>
+```
+
+> [!NOTE]
+> For additional details please see libvirt's [documentation](https://libvirt.org/formatdomain.html#modifying-virtual-link-state).
+
+virt-launcher's `LibvirtDomainManager.SyncVMI` will be extended to support updating the interfaces link state while the VM is running.
+
+virt-launcher's NIC hotplug logic will be adjusted to support hot-plugging a NIC while its link state is down.
+
+## Network Binding Plugins Support
+
+When using the functionality purposed by this proposal with network binding plugins using a sidecar container -
+its up to the plugin's vendor to support this functionality - as they control the interface section in the domain XML.
+
+## Link State Reporting
+
+A new field will be added to the VMI interface status that will be used for reporting the interface's current link state (up/down):
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+	...
+	LinkState string `json:"linkState,omitempty"`
+	...
+}
+```
+
+virt-handler will be responsible for reporting this field's value, based on domain information received from virt-launcher.
+
+## API Examples
+### Controlling The Link State
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+            - name: default
+              state: linkDown
+              masquerade: { }
+      networks:
+        - name: default
+          pod: { }
+```
+### Link State Reporting
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vm
+spec:
+  domain:
+    devices:
+      interfaces:
+      - name: default
+        state: linkDown
+        masquerade: { }
+  networks:
+  - name: default
+    pod: { }
+status:
+  interfaces:
+    - name: default
+      linkState: down
```

Comment:
Requesting `linkDown` but getting `down` is also confusing IMHO.

Reply:
Adjusted the allowed values to `up` and `down`.